packages/ui-lib/src/Filters/atoms/FilterDropHandler.tsx

```diff
@@ -84,77 +84,105 @@ interface IFilterDropHandler {
-  const mainRef = useRef<HTMLDivElement>(null);
-
-  const handleClose = useCallback(() => {
-    if (openState.isOpen) {
-      onCloseCallback();
-    }
-
-    setOpenState((prev) => {
-      const isOpen = false;
-      return { ...prev, isOpen };
-    });
-
-  }, [onCloseCallback, openState.isOpen]);
-
-  useClickOutside(mainRef, handleClose);
-
-  const handleToggleOpen = useCallback((minWidth: number, minHeight: number) => {
-    if (!buttonWrapperRef.current) { return; }
-
-    const buttonRect = buttonWrapperRef.current.getBoundingClientRect();
-    if (!buttonRect) { return; }
-    const position: IOpenPos = getDropPosition(buttonRect, minWidth, minHeight);
+export interface FilterDropHandlerRef {
+  cancelClose: () => void;
+}
 
-    onToggleOpenCallback(openState.isOpen);
-    setOpenState((prev) => {
-      const isOpen = !prev.isOpen;
-      return { ...prev, isOpen, position };
+const FilterDropHandler = forwardRef<FilterDropHandlerRef, IFilterDropHandler>(
+  (
+    {
+      buttonIcon,
+      buttonText,
+      disabled = false,
+      minWidth = 270,
+      minHeight = 190,
+      isSortAscending,
+      design = 'default',
+      noCloseOnClickOutside,
+      children,
+      onToggleOpenCallback = () => { },
+      onCloseCallback = () => { },
+      ...props
+    },
+    imperativeRef
+  ) => {
+
+    const [openState, setOpenState] = useState<IDropOpen>({
+      isOpen: false,
+      position: 'bottom-right',
     });
-  }, [onToggleOpenCallback, openState.isOpen]);
-
-  return (
-    <Container ref={mainRef} {...props}>
-      <ButtonWrapper ref={buttonWrapperRef}>
-        <FilterButton
-          icon={buttonIcon}
-          isOpen={openState.isOpen}
-          onClick={() => handleToggleOpen(minWidth, minHeight)}
-          {...{ disabled, isSortAscending, design }}
-          hasFlipArrow
-        >{buttonText}
-        </FilterButton>
-      </ButtonWrapper>
-      <ContentBox {...{ openState, disabled, minWidth }}>
-        {children}
-      </ContentBox>
-    </Container>
-  );
-};
+
+    const buttonWrapperRef = useRef<HTMLDivElement>(null);
+    const mainRef = useRef<HTMLDivElement>(null);
+
+    const clickOutsideClose = useCallback(() => {
+      if(noCloseOnClickOutside) {
+        return;
+      }
+
+      if (openState.isOpen) {
+        onCloseCallback();
+      }
+
+      setOpenState((prev) => {
+        const isOpen = false;
+        return { ...prev, isOpen };
+      });
+
+    }, [noCloseOnClickOutside, onCloseCallback, openState.isOpen]);
+
+    useClickOutside(mainRef, clickOutsideClose);
+
+    const handleToggleOpen = useCallback((minWidth: number, minHeight: number) => {
+      if (!buttonWrapperRef.current) { return; }
+
+      const buttonRect = buttonWrapperRef.current.getBoundingClientRect();
+      if (!buttonRect) { return; }
+      const position: IOpenPos = getDropPosition(buttonRect, minWidth, minHeight);
+
+      onToggleOpenCallback(!openState.isOpen);
```

Comment:
I corrected the indentation of this file so it's a bit confusing, there are 3 updates in this file

I have corrected the logic here on the handleToggleOpen, previously was sending the previous state of openState, now I'm sending the updated state. This is reflected in the FilterBar update that sends the callback when isOpen is false.

onClickOutsideClose  previously named handle close it will review noCloseOnClickOutside, since the FilterDropdownHandler only closes onClickOutside.

I have updated the Component to have forward reference to add the cancelClose functionality from outside.